### PR TITLE
fix: histogram only on non-nan values

### PR DESCRIPTION
**Pull request by sahahner**
Fix issue #14

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=15) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=15) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-training?pullRequest=15) it.</sub>

**Review by anaprietonem — APPROVED** [review]
This PR looks to me! As a minor thing I would add a comment to explain that because of the postprocessors being applied, then the data might contains NaNs

**Comment by sahahner** [comment]
In addition to fixing the issue, I noticed that the histogram plot containing histograms for the true and predicted variable does not fix the range of the bins. This leads to bins of different sizes and at different positions. Therefore, the histograms do not allow a fair comparison of the two variables if the values lie in different ranges.
The last commit incorporates a change, which forces the two variables to be binned into the same bins.

The effect on the histogram plotted after few epochs, when the differences are still high, is clearly visible:
Before: with bins covering smaller ranges on the x-axis, densities of true data go up to 3500
image
After: density up to 8000 using the same bins for both 
image

I also changed the name of the label "Truth (ERA5)" to "Truth (data)"

@anaprietonem, can you please have a look at the additional changes?

**Comment by anaprietonem** [comment]
> In addition to fixing the issue, I noticed that ...
So If I am reading correctly what you are doing is passing the range to the histogram so when plotting both the histogram for the predicted and truth data that uses the same range, right?

> I also changed the name of the label "Truth (ERA5)" to "Truth (data)"
About this, so I assume you'd like to remove ERA5 since we might use other training datasets? If so maybe we should change it also in the spectra plot where there is another label like that. What do you think about using 'Predicted' or 'AI-predicted' rather than anemoi?

**Comment by sahahner** [comment]
> So If I am reading correctly what you are doing is passing the range to the histogram so when plotting both the histogram for the predicted and truth data that uses the same range, right?

Yes.

**Comment by sahahner** [comment]
> About this, so I assume you'd like to remove ERA5 since we might use other training datasets? If so maybe we should change it also in the spectra plot where there is another label like that. What do you think about using 'Predicted' or 'AI-predicted' rather than anemoi?

That is a very good proposal. I changed it in the other plot as well.

Yes, ERA5 is not necessarily used by everyone. I also want to point out that this is not the meteorological truth, but what our model considers as true data.

**Review by anaprietonem — DISMISSED** [review]
The changes applied to the histogram function looks good to me!

**Review by gmertes — APPROVED** [review]
